### PR TITLE
Fix iOS and Android bugs

### PR DESCRIPTION
**Pull request by danieldunderfelt**
Hi!

This PR fixes both #21 and #19. On the Android side `!outputPath.isEmpty()` caused a Null Pointer Exception if outputPath is null (might want to use &&, this is not tested with an outputPath). On the iOS side I started getting nasty errors after switching the deprecated `loadImageWithTag` method to `loadImageWithURLRequest` that I tracked down to the new method wanting an `NSURLRequest` object instead of a string. I fixed this with the help of StackOverflow, but as I am not an objective-c developer you might want to look over the code I added to transform the path string into a request object.

In short, with this PR the resizer works perfectly for my use cases, without using deprecated methods, on both iOS and Android running React-Native 0.29.

### CODE REVIEW

**Comment by stoffern** [comment]
This looks good, can be merged :+1: 

cc @4ian

**Comment by 4ian** [comment]
Hi @danieldunderfelt and @stoffern ! 
Thanks for the PR. 😄  Unfortunately, this is breaking compatibility with older version of React Native on the iOS side, as `loadImageWithURLRequest` is not present. I might merge it anyway but the example app needs to be updated (right now it's using RN 0.25, which do not have `loadImageWithURLRequest`).

**Comment by stoffern** [comment]
@4ian, it is bad practice to have a example app with all sourcecode for android and ios.
You will need to upgrade them every 2. week when RN updates. That is not good.

Just link the example.js file and in readme put:
```
#example 
npm i -S react-native-image-resizer
import ImageResizer from './node_modules/react-native-image-resizer/example/'
```
I can fix it for you :)

**Comment by 4ian** [comment]
I'm not sure why this would be a bad practise, I mean on the contrary it's quite convenient for new developers that want to try the library: they just have to clone the git, do a `npm install` in the example folder and start the app to be able to test it :)
Also for me it's quite handy as I'm testing the library with this.

> You will need to upgrade them every 2. week when RN updates. That is alot of work.

The version of React Native is fixed in the package.json of the example app, so no need to update it. Of course when it's getting too old it should be updated, but that depends on what version we want to support in this library :)

**Comment by stoffern** [comment]
@4ian well, then why not just add the package.json, and let the user himself get the files that is needed (node_modules, android and ios). so for next "react-native upgrade" of the example app you just change version in the package.json

Installing the example will then be:
```
- npm i
- react-native upgrade
- react-native link
- react-native start
```

**Comment by 4ian** [comment]
The `NullPointerException` issue and the deprecated use of `loadImageWithTag` were both fixed in other PRs, so I'm closing this one ;)
